Apps/Common/src/AccessManagement/Authentication/AuthenticationDelegate.cs

```diff
@@ -0,0 +1,162 @@
+//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.Common.AccessManagement.Authentication
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using HealthGateway.Common.AccessManagement.Authentication.Models;
+    using HealthGateway.Common.Services;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+    using System.Text.Json;
+
+    /// <summary>
+    /// The Authorization service.
+    /// </summary>
+    public class AuthenticationDelegate : IAuthenticationDelegate
+    {
+        const string  CONFIGSECTIONNAME = "ClientAuthentication";
+        private readonly ILogger<AuthenticationDelegate> logger;
+
+        private readonly IHttpClientService httpClientService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthenticationDelegate"/> class.
+        /// </summary>
+        /// <param name="logger">The injected logger provider.</param>
+        /// <param name="config">The configuration.</param>
+        /// <param name="httpClientService">The injected http client service.</param>
+        public AuthenticationDelegate(
+            ILogger<AuthenticationDelegate> logger,
+            IConfiguration config,
+            IHttpClientService httpClientService)
+        {
+            this.logger = logger;
+            this.httpClientService = httpClientService;
+            IConfigurationSection? configSection = config?.GetSection(CONFIGSECTIONNAME);
+
+            this.TokenUri = new Uri(configSection.GetValue<string>("TokenUri"));
+            this.TokenRequest = new ClientCredentialsTokenRequest()
+            {
+                Audience = configSection.GetValue<string>("Audience"),
```

Comment:
you should probably model this and use binding instead.

Reply:
can defer this change. Need to create a model.